pkg/pool-weighted/contracts/utils/IndexPoolUtils.sol

```diff
@@ -82,4 +84,33 @@ contract IndexPoolUtils {
 
         return normalizedWeights;
     }
+
+    /// @dev Calculates the weight to be used to calculate the price of an uninitialized token depending on its amount in pool.
+    /// @param _newTokenBalanceIn Amount of uninitialized token in pool (after the swap)
+    /// @param _minimumBalance Minimum balance set for the uninitialized token (= initialization threshold)
+    /// @return Weight to be used to calculate the price of an uninitalized token.
+    function _getUninitializedTokenWeight(uint256 _newTokenBalanceIn, uint256 _minimumBalance)
+        internal
+        view
+        returns (uint256)
+    {
+        uint256 currentWeight = HUNDRED_PERCENT / 100;
+
+        bool addPremium = _newTokenBalanceIn < _minimumBalance;
```

Comment:
Hmm, I suppose if we are swapping in 1,1% of newTokens we should still receive premium for 1%, and have 0,1% without premium, no?

Reply:
Maybe I'm overcomplicating tbh...

Reply:
But this way if I try to sap exactly 1% in I won't get any incentives... :(

Reply:
You are raising a good point here. The documentation is a bit ambiguous here, but after looking at the Indesx v1 code, it seems to me as if we should be using the tokenBalanceIn **before** the swap has occured, which would make more sense and account for the cases that you mentioned.